addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -105,7 +105,7 @@ def _compute_project_id(self):
     @api.depends('project_id')
     def _compute_task_id(self):
         for line in self:
-            if line.project_id and line.project_id == line.task_id.project_id:
+            if line.project_id:
                 continue
             line.task_id = False
```

Comment:
Hu, compute methods are a little weird in this file...
```suggestion
            if not line.project_id:
                line.task_id = False
```
or this is even better, because we would write on all tasks at once :
```
        self.filtered(lambda t: not t.project_id).task_id = False
```